internal/payout/transaction.go

```diff
@@ -0,0 +1,201 @@
+package payout
+
+import (
+	gsrpc "github.com/NodeFactoryIo/go-substrate-rpc-client"
+	"github.com/NodeFactoryIo/go-substrate-rpc-client/signature"
+	"github.com/NodeFactoryIo/go-substrate-rpc-client/types"
+	log "github.com/sirupsen/logrus"
+	"math/big"
+	"sync"
+)
+
+type TransactionStatus string
+
+const (
+	Finalized = TransactionStatus("Finalized")
+	Dropped   = TransactionStatus("Dropped")
+	Invalid   = TransactionStatus("Invalid")
+)
+
+type TransactionDetails struct {
+	NodeId string
+	To     string
+	Amount big.Int
+	Status TransactionStatus
+}
+
+func ExecuteAllPayoutTransactions(
+	payoutDistribution map[string]big.Int,
+	payoutDetails map[string]NodePayoutDetails,
+	secret string,
+	substrateRPCurl string,
+) ([]*TransactionDetails, error) {
+	api, err := gsrpc.NewSubstrateAPI(substrateRPCurl)
```

Comment:
Can you refactor these functions into separate functions to make it more clear.
Bit hard to read right now
Eg. InitializeApi, ...

Reply:
I am not sure I understand, did you mean something like this (? 

```go
func InitializeApi(substrateRPCurl string) (*gsrpc.SubstrateAPI, error) {
	api, err := gsrpc.NewSubstrateAPI(substrateRPCurl)
	return api, err
}
```